pkg/envoy/bootstrap/config.go

```diff
@@ -330,43 +331,106 @@ func GetValidationContextSDSConfigYAML() ([]byte, error) {
 // startup, readiness and liveness probes.
 // These will not change during the lifetime of the Pod.
 // If the original probe defined a TCPSocket action, listener and cluster objects are not configured
-// to serve that probe.
+// to serve that probe since the osm-healthcheck container that does that job is configured separately.
 func (b *Builder) getProbeResources() ([]*xds_listener.Listener, []*xds_cluster.Cluster, error) {
 	// This slice is the list of listeners for liveness, readiness, startup IF these have been configured in the Pod Spec
 	var listeners []*xds_listener.Listener
 	var clusters []*xds_cluster.Cluster
 
-	// Is there a liveness probe in the Pod Spec?
-	if b.OriginalHealthProbes.Liveness != nil && !b.OriginalHealthProbes.Liveness.IsTCPSocket {
-		listener, err := getLivenessListener(b.OriginalHealthProbes.Liveness)
-		if err != nil {
-			log.Error().Err(err).Msgf("Error getting liveness listener")
-			return nil, nil, err
-		}
-		listeners = append(listeners, listener)
-		clusters = append(clusters, getLivenessCluster(b.OriginalHealthProbes.Liveness))
+	livenessListenerBuilder := probeListenerBuilder{
+		listenerName: livenessListener,
+		inboundPort:  constants.LivenessProbePort,
+	}
+	readinessListenerBuilder := probeListenerBuilder{
+		listenerName: readinessListener,
+		inboundPort:  constants.ReadinessProbePort,
+	}
+	startupListenerBuilder := probeListenerBuilder{
+		listenerName: startupListener,
+		inboundPort:  constants.StartupProbePort,
 	}
 
-	// Is there a readiness probe in the Pod Spec?
-	if b.OriginalHealthProbes.Readiness != nil && !b.OriginalHealthProbes.Readiness.IsTCPSocket {
-		listener, err := getReadinessListener(b.OriginalHealthProbes.Readiness)
-		if err != nil {
-			log.Error().Err(err).Msgf("Error getting readiness listener")
-			return nil, nil, err
+	for containerName, probes := range b.OriginalHealthProbes {
+		// Is there a liveness probe in the Pod Spec?
+		if probes.Liveness != nil && !probes.Liveness.IsTCPSocket {
+			clusterName := fmt.Sprintf("%s_%s", containerName, livenessCluster)
+			if probes.Liveness.IsHTTP {
+				livenessListenerBuilder.isHTTP = true
```

Comment:
Should this have isHTTP set globally on the livesnessBuilder, even if there are others that are no http?

Reply:
Not sure what you mean; there are up to 3 different builders representing the different probes that exist in Kubernetes. As such, each builder has an `IsHTTP` field

Reply:
is it possible for there to be more than 1 liveness builder in the list of `range b.OriginalHealthProbes`?

If so, the isHTTP is a global field, even if there is one HTTP liveness, and one TCP liveness

Reply:
to clarify, i'm suggesting the code turns into:


```go
for containerName, probes := range b.OriginalHealthProbes {
  livenessBuilder.AddProbe(probes.Liveness)
  readinessBuilder.AddProbe(probes.Readiness)
  startupBuilder.AddProbe(probes.Startup)
}
```

You can also add the type and the container name as fields to the model.Probe, or simply add extra params to the AddProbe method

Reply:
Ah, you're absolutely right! Great catch; what I'll do is add a TLS inspector so that Envoy can detect HTTPS traffic